### ISSUE TITLE
[Bug]: Layout route with more than one pathless layout causes invalid conflict

### ISSUE THREAD

**Issue by pokedotdev** [remix-run/remix#3327]
### What version of Remix are you using?

1.5.1

### Steps to Reproduce

Create a route structure like so:
```
app/routes
├── slug
│  ├── __bar
│  │  └── bar.tsx
│  ├── __foo
│  │  ├── foo.tsx
│  │  └── index.tsx
│  ├── __bar.tsx
│  └── __foo.tsx
└── slug.tsx
```

### Expected Behavior

Layout routes can have more than one pathless layout route.

### Actual Behavior

The error is thrown when starting remix:
```
Error: Path "slug/" defined by route "routes/slug/__foo" conflicts with route "routes/slug/__bar"
```
[You can see reproduction here](https://stackblitz.com/edit/github-3s7qm8?file=app/routes/$slug.tsx)

**Comment by goldo** [comment]
Hi, I can confirm this, I have the same problem. (I wouldn't flag it as "feat" tho, it looks like a bug to me)

**Comment by binajmen** [comment]
I have the same problem. @machour, could you flag it as bug instead?

**Comment by machour** [comment]
Will be fixed by #4421

**Comment by brophdawg11** [comment]
#4421 is merged and should be available once `v1.17.0` is released
